packages/gatsby-cli/src/init-starter.ts

```diff
@@ -115,7 +101,10 @@ const install = async (rootPath: string): Promise<void> => {
   process.chdir(rootPath)
 
   try {
-    if (await shouldUseYarn()) {
+    if (!getPackageManager()) {
+      setPackageManager(`npm`)
+    }
+    if (getPackageManager() === `yarn` && checkForYarn()) {
       await fs.remove(`package-lock.json`)
```

Comment:
This would convert a package lock to a yarn.lock and packages will be the same as with npm.
```suggestion
      if (fs.fileExists(`package-lock.json`) {
        await spawn('yarnpkg import')
        await fs.remove(`package-lock.json`)
      }
```